src/request.ts

```diff
@@ -336,13 +330,12 @@ class DatastoreRequest {
    *   const apiResponse = data[0];
    * });
    */
-  delete(keys, gaxOptions?, callback?) {
-    if (is.fn(gaxOptions)) {
-      callback = gaxOptions;
-      gaxOptions = {};
-    }
-
-    callback = callback || (() => {});
+  delete(keys:Entities): void;
+  delete(keys: Entities, callback: google.datastore.v1.Datastore.CommitCallback): void;
+  delete(keys: Entities, gaxOptions: CallOptions, callback: google.datastore.v1.Datastore.CommitCallback): void;
+  delete(keys: Entities, gaxOptionsOrCallback?: CallOptions|google.datastore.v1.Datastore.CommitCallback, callback?: google.datastore.v1.Datastore.CommitCallback): void {
+    const gaxOptions = typeof gaxOptionsOrCallback === 'object' ? gaxOptionsOrCallback : {};
+    callback = typeof gaxOptionsOrCallback === 'function' ? gaxOptionsOrCallback : (() => {});
```

Comment:
Should be `: callback;`, not `(() => {})`

Reply:
Another lint bypass for this error: `[ts]
Argument of type 'CommitCallback | undefined' is not assignable to parameter of type 'RequestCallback'.
  Type 'undefined' is not assignable to type 'RequestCallback'. [2345]` - Fixed!